app/src/main/java/com/duckduckgo/app/email/EmailJavascriptInterface.kt

```diff
@@ -17,35 +17,40 @@
 package com.duckduckgo.app.email
 
 import android.webkit.JavascriptInterface
-import timber.log.Timber
+import android.webkit.WebView
+import com.duckduckgo.app.browser.DuckDuckGoUrlDetector
+import com.duckduckgo.app.global.DispatcherProvider
+import kotlinx.coroutines.runBlocking
 
 class EmailJavascriptInterface(
     private val emailManager: EmailManager,
-    private val showNativeTooltip: () -> Unit
+    private val showNativeTooltip: () -> Unit,
+    private val webView: WebView,
+    private val urlDetector: DuckDuckGoUrlDetector,
+    private val dispatcherProvider: DispatcherProvider
 ) {
 
-    @JavascriptInterface
-    fun log(message: String) {
-        Timber.i("EmailInterface $message")
+    private fun isUrlFromDuckDuckGoEmail(): Boolean {
+        return runBlocking(dispatcherProvider.main()) {
+            val url = webView.url
+            (url != null && urlDetector.isDuckDuckGoEmailUrl(url))
+        }
     }
 
     @JavascriptInterface
-    fun getAlias(): String {
-        val nextAlias = emailManager.getAlias()
-
-        return if (nextAlias.isNullOrBlank()) {
-            ""
+    fun isSignedIn(): String {
```

Comment:
nit: this is an odd method name, `isSignedIn` suggest it will return a `boolean` but returns a string

Reply:
Yeah, I can't fully remember but I think this is because we can only pass strings back to the JS code.